sorting/InPlaceMergeSort.c

```diff
@@ -0,0 +1,119 @@
+#include<stdio.h>
+#include<stdlib.h>
+
+void swap(int * x, int * y )
+{
+	int z = *x;
+	*x = *y;
+	*y = z;
+}
+
+
+void insertionSort(int *arr, int s, int e)
+{
+   int i, key, j;
+   for (i = s+1; i < e; i++)
+   {
+       key = arr[i];
+       j = i-1;
+       while (j >= 0 && arr[j] > key)
+       {
+           arr[j+1] = arr[j];
+           j = j-1;
+       }
+       arr[j+1] = key;
+   }
+}
+
+
+
+void concat(int * arr, int l , int m, int r)
+{
+	int lh = m-l + 1 ,  rh = r-m ;
+	int * brr = malloc(sizeof(int)*lh) , * crr = malloc(sizeof(int)*rh) ;
```

Comment:
if you aren't using ``brr`` and ``crr`` why are you allocating memory to it? And you are not releasing the memory, causing memory leaks.